env0/resource_notification_project_assignment_test.go

```diff
@@ -0,0 +1,193 @@
+package env0
+
+import (
+	"errors"
+	"regexp"
+	"testing"
+
+	"github.com/env0/terraform-provider-env0/client"
+	"github.com/golang/mock/gomock"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestUnitNotificationProjectAssignmentResource(t *testing.T) {
+	resourceType := "env0_notification_project_assignment"
+	resourceName := "test"
+	accessor := resourceAccessor(resourceType, resourceName)
+
+	projectId := "pid1"
+
+	assignment := client.NotificationProjectAssignment{
+		Id:                     "id1",
+		NotificationEndpointId: "nid1",
+		EventNames: []string{
+			"driftDetected",
+		},
+	}
+
+	assignmentEventNamesUpdated := client.NotificationProjectAssignment{
+		Id:                     "id1",
+		NotificationEndpointId: "nid1",
+		EventNames: []string{
+			"driftDetected", "environmentDestroyStarted",
+		},
+	}
+
+	assignmentNotificationEndpointIdUpdated := client.NotificationProjectAssignment{
+		Id:                     "id2",
+		NotificationEndpointId: "nid2",
+		EventNames: []string{
+			"driftDetected",
+		},
+	}
+
+	t.Run("Success", func(t *testing.T) {
+		testCase := resource.TestCase{
+			Steps: []resource.TestStep{
+				{
+					Config: resourceConfigCreate(resourceType, resourceName, map[string]interface{}{
+						"project_id":               projectId,
+						"notification_endpoint_id": assignment.NotificationEndpointId,
+						"event_names":              []string{"driftDetected"},
+					}),
+					Check: resource.ComposeAggregateTestCheckFunc(
+						resource.TestCheckResourceAttr(accessor, "id", assignment.Id),
+						resource.TestCheckResourceAttr(accessor, "project_id", projectId),
+						resource.TestCheckResourceAttr(accessor, "notification_endpoint_id", assignment.NotificationEndpointId),
+						resource.TestCheckResourceAttr(accessor, "event_names.#", "1"),
```

Comment:
Took my a while to figure this one out:
.# - counts the elements.
.* - access element in index *